### DISCUSSION TITLE
JS Manipulation Search Result Webpart (v3.23)

### DISCUSSION THREAD

**Discussion by Heavm00** [microsoft-search/pnp-modern-search#2338] [Q&A]
Hi @wobba,
I'm using the Modern Script Editor Webpart for manipulation of the search results (re-write url from people delve profile to bing profiles, ...).
The thing is, everythings works perfect onload. When now someone enters anything in my textbox, the search results are re-loaded (without page refresh). Which event could I use, to trigger my JS ?

That is was I'm trying so far, but without any luck:

```
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
<script>
    $(function() {
        
        $("[class^=searchButton_]").click(function() {
          ChangePersonRedirect();  // --> fired, but not working
        });
        
        $("[id^=SearchBox]").on("keypress", function(e) {
            if (e.keyCode == 13) {
                alert("Enter pressed");  // --> not fired
                return false; // prevent the button click from happening
            }
        });
        
        setTimeout(ChangePersonRedirect, 1000); // --> this works, url is set correct
        
    });
    
    // change url from delve to bing profile for people results
    function ChangePersonRedirect() {
        
        $('.ms-Persona-primaryText').each(function(i, obj) {
            
            var oldUrl = $(this).find("a").attr("href");
            var pos = oldUrl.lastIndexOf("|");
            var email = oldUrl.substr(pos+1);
            
            var newUrl = "https://www.bing.com/work/search?q=" + email; 
            $(this).find("a").attr("href", newUrl); // Set herf value
        });
    }
</script>
```

If this is not working I'll catch your code and do the modifications within Webpart code.

Would be thankful for any help. Thanks!

**Comment by wobba** [comment]
Can you explain a bit more on what you are trying to achieve? Is it rewrite the clicks on OOB people results when you search from SharePoint to pass them over to Bing@Work, showing the People answer in that experience? Or do you want to change the PnP search box behavior?

And, using the script editor webpart is likely not the way to go about this 🙂

**Reply by Heavm00** [reply]
Hi,
thanks for response.  I'm using pnp search box in combination with pnp search results for creating an employee list. Primary text ist linked to person details automatically (delve profile). The only thing I want to to, is, to re-write the URL, so that's linked to www.bing.com/work/[users e-mail-adress]. I cannot do that with handlebar expression, becaue the html is not rendered correct (html text is shon).

**Reply by wobba** [reply]
You should be able to edit the handlebars template to do what you want. Which layout are you using? and I'll see if I can provide a sample for you.

**Reply by Heavm00** [reply]
I'm using the people template, where primary text is a link to the user profile.
![grafik](https://user-images.githubusercontent.com/61208008/188636063-c56aa1fc-1cea-4881-becb-b530141ad185.png)

How can I change that URL within the Persona Fields?
![grafik](https://user-images.githubusercontent.com/61208008/188636690-5e09fc0c-c486-4a2c-8972-262e2830be0c.png)
The URL is not renderd here. I would like to render the following way:  `<a href="https://www.bing.com/work/{{WorkEmail}}">{{LastName}} {{FirstName}}</a>`

**Reply by wobba** [reply]
Which web part version are you using?

**Reply by Heavm00** [reply]
v3.23

**Reply by wobba** [reply]
Hi, there's no easy way to do this using v3. I'd move to v4 rather than add script to try to update the URL. I'll add this one as an enhancement to do if I have time, to expose the URL property as well in the field mapping.
